floors/rule.go

```diff
@@ -0,0 +1,552 @@
+package floors
+
+import (
+	"fmt"
+	"math"
+	"math/bits"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/currency"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	SiteDomain          string = "siteDomain"
+	PubDomain           string = "pubDomain"
+	Domain              string = "domain"
+	Bundle              string = "bundle"
+	Channel             string = "channel"
+	MediaType           string = "mediaType"
+	Size                string = "size"
+	GptSlot             string = "gptSlot"
+	AdUnitCode          string = "adUnitCode"
+	Country             string = "country"
+	DeviceType          string = "deviceType"
+	Tablet              string = "tablet"
+	Desktop             string = "desktop"
+	Phone               string = "phone"
+	BannerMedia         string = "banner"
+	VideoMedia          string = "video-instream"
+	VideoOutstreamMedia string = "video-outstream"
+	AudioMedia          string = "audio"
+	NativeMedia         string = "native"
+)
+
+type ruleKeys struct {
+	keyMap    map[string]bool
+	keys      []string
+	delimiter string
+}
```

Comment:
Can you move this type declaration so it is with its constructor function and its receiver methods?

Reply:
moved down